vhd/lib/libvhd.c

```diff
@@ -753,8 +737,10 @@ void
 vhd_bitmap_set(vhd_context_t *ctx, char *map, uint32_t block)
 {
 	if (vhd_creator_tapdisk(ctx) &&
-	    ctx->footer.crtr_ver == 0x00000001)
-		return old_set_bit(map, block);
+	    ctx->footer.crtr_ver == 0x00000001) {
+		VHDLOG("error unsupported version 0.1 VHD");
+		abort();
+	}
```

Comment:
IMO this ought to be in vhd_open() and to return an API error rather than coredumping.

As a bonus that removes logic from what is clearly a semi-fast (if not fully fast) path.